xclim/core/indicator.py

```diff
@@ -313,10 +364,29 @@ def __call__(self, *args, **kwds):
         ba = self._sig.bind(*args, **kwds)
         ba.apply_defaults()
 
+        # Get inputs passed as strings from ds
```

Comment:
I suggest moving this logic into a method, e.g. `check_named_args` or something like that.